modules/core/src/main/scala/schema.scala

```diff
@@ -23,12 +23,57 @@ trait Schema {
 
   def pos: SourcePos
 
-  /** The types defined by this `Schema`. */
-  def types: List[NamedType]
+  /** The types defined by this `Schema` prior to any extensions. */
+  def unExtendedTypes: List[NamedType]
+
+  /** The types defined by this `Schema` with any extensions applied. */
+  def types: List[NamedType] = {
+    unExtendedTypes.map { tpe =>
+      val exts = extensions.filter(_.extended.dealias =:= tpe)
+      if (exts.length == 0) {tpe} else { extendType(tpe, exts) }
+    }
+  }
+
+  private def extendType(extendee: NamedType, toApply: List[NamedTypeExtension]): NamedType = {
+    extendee match {
+        case TypeRef(_, _) => throw new RuntimeException("Invariant violated")
```

Comment:
Not sure if this is the correct thing to do here?

Reply:
Could we accommodate this case with `extendee.dealias match { ... }`?

Reply:
Compiler still complains about the match not being exhaustive